### PR TITLE
NMS-12431: Allow selecting the layer via URL parameter in topology ui

### PR DESCRIPTION
**Pull request by mvrueden**
JIRA: https://issues.opennms.org/browse/NMS-12431

### CODE REVIEW

**Comment by agalue** [comment]
Excellent news, thank you for working on this~

Although the customer who requested this functionality is running Meridian 2018, and upgrading their environment can be challenging and could require more planning and testing from their part.

Is it possible to rebase the solution for Meridian 2018? I'd like to have an answer before updating the customer with the options they have.

File: features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/TopologyUI.java [hunk 1]
```diff
@@ -206,11 +208,26 @@ private TopologyUIRequestHandler() {
                     // The order matters
                     this::loadHistoryFragment,
                     this::loadGraphProvider,
+                    this::loadLayer,
                     this::loadVertexHopCriteria,
                     this::loadSemanticZoomLevel,
                     this::loadLayout);
         }
 
+        private boolean loadLayer(VaadinRequest request) {
```
Comment (patrick-schweizer) [inline features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/TopologyUI.java]: nit pick: there are quite a few conditions within conditions in this code block. But I guess there is no nicer easy way to express this....

**Comment by mvrueden** [comment]
@agalue the changes here need to be manually backported to `foundation-2018` which I will do in a separate PR.

**Comment by agalue** [comment]
Excellent, thank you!

**Pull request merged**